code/game/objects/items/cosmetics.dm

```diff
@@ -131,60 +131,100 @@
+					if(!get_location_accessible(H, location))
+						to_chat(user, "<span class='warning'>The mask is in the way!</span>")
+						return
+					user.visible_message("<span class='notice'>[user] tries to change [H]'s facial hair style using [src].</span>", "<span class='notice'>You try to change [H]'s facial hair style using [src].</span>")
+					if(new_style && do_after(user, 60, target = H))
+						user.visible_message("<span class='notice'>[user] successfully changes [H]'s facial hair style using [src].</span>", "<span class='notice'>You successfully change [H]'s facial hair style using [src].</span>")
+						H.facial_hair_style = new_style
+						H.update_hair()
+
+					return
+				else
+					return
 
-			if(H == user) //shaving yourself
-				user.visible_message("[user] starts to shave [user.p_their()] facial hair with [src].", \
-									 "<span class='notice'>You take a moment to shave your facial hair with [src]...</span>")
-				if(do_after(user, 50, target = H))
-					user.visible_message("[user] shaves [user.p_their()] facial hair clean with [src].", \
-										 "<span class='notice'>You finish shaving with [src]. Fast and clean!</span>")
-					shave(H, location)
 			else
-				var/turf/H_loc = H.loc
-				user.visible_message("<span class='warning'>[user] tries to shave [H]'s facial hair with [src].</span>", \
-									 "<span class='notice'>You start shaving [H]'s facial hair...</span>")
-				if(do_after(user, 50, target = H))
-					if(H_loc == H.loc)
-						user.visible_message("<span class='warning'>[user] shaves off [H]'s facial hair with [src].</span>", \
-											 "<span class='notice'>You shave [H]'s facial hair clean off.</span>")
+				if(!(FACEHAIR in H.dna.species.species_traits))
+					to_chat(user, "<span class='warning'>There is no facial hair to shave!</span>")
+					return
+				if(!get_location_accessible(H, location))
+					to_chat(user, "<span class='warning'>The mask is in the way!</span>")
+					return
+				if(H.facial_hair_style == "Shaved")
+					to_chat(user, "<span class='warning'>Already clean-shaven!</span>")
+					return
+
+				if(H == user) //shaving yourself
+					user.visible_message("[user] starts to shave [user.p_their()] facial hair with [src].", \
+										 "<span class='notice'>You take a moment to shave your facial hair with [src]...</span>")
+					if(do_after(user, 50, target = H))
+						user.visible_message("[user] shaves [user.p_their()] facial hair clean with [src].", \
+											 "<span class='notice'>You finish shaving with [src]. Fast and clean!</span>")
 						shave(H, location)
+				else
+					var/turf/H_loc = H.loc
+					user.visible_message("<span class='warning'>[user] tries to shave [H]'s facial hair with [src].</span>", \
+										 "<span class='notice'>You start shaving [H]'s facial hair...</span>")
+					if(do_after(user, 50, target = H))
+						if(H_loc == H.loc)

```

Comment:
isn't this already handled by do_after?

Reply:
https://github.com/tgstation/tgstation/blob/513944c8daff765330ffb7173c2882e119dbd58d/code/__HELPERS/mobs.dm#L277

Reply:
I didnt actually write that piece of code but yeah I think so

If you want me to fix it it'll have to wait for after the weekend as i'm away from home